test/expect/TestCollectEnv.test_pytorch_win_ws2016_cuda9_cudnn7_py3.expect

```diff
@@ -0,0 +1,18 @@
+PyTorch version: 0.4.0a0
+Is debug build: No
+CUDA used to build PyTorch: 9.0
+
+OS: Microsoft Windows Server 2012 R2 Standard
+GCC version: Could not collect
```

Comment:
At some point it would be good to collect the MSVC version. Maybe @peterjc123 can help.

Reply:
My implementation can be found [here](https://github.com/pytorch/pytorch/blob/master/torch/utils/cpp_extension.py#L73). Make sure you wrap it with a try-catch block, since the compiler may be not available if the user doesn't activate the dev shell.

Reply:
Thanks @peterjc123! I'll take a look